#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/festo/robotino3/protos/Robotino3Webcam.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/fruits/protos/Apple.proto"

WorldInfo {
  gravity 9.18
}
Viewpoint {
  orientation 0.5458827744418958 0.5414428717583019 -0.639415055491925 2.0111914987072153
  position 0.16309161797077418 0.3680127199873555 3.1215289946686444
}
DEF Body Pose {
}
RectangleArena {
  floorSize 2 2
  floorAppearance PBRAppearance {
    transparency 1
  }
  wallAppearance PBRAppearance {
    transparency 1
  }
}
DEF ROB Robot {
  translation 0 0.5 0
  rotation 0 0 -1 1.5707996938995747
  children [
    InertialUnit {
      translation -0.01 0 0
      rotation 0 0 -1 0
      name "imu"
    }
    Lidar {
      translation 0.01 0 0.11
      fieldOfView 0.7
      verticalFieldOfView 0.7
      numberOfLayers 50
    }
    Compass {
    }
    GPS {
    }
    RangeFinder {
      translation 0.09 0.015 0.1
    }
    DEF Body Pose {
      translation 0 0 0.0415
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    Robotino3Webcam {
      translation -0.1 0 -0.2
      rotation 0 0 1 1.5708003061004252
      name "camera"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 600.5169884160529
        anchor 0.045 0 0.025
      }
      device [
        RotationalMotor {
          name "motor_1"
        }
      ]
      endPoint Solid {
        translation 0.045073090380146286 2.1504310780816584e-05 0.02497597286746378
        rotation -0.6967347685173485 0.16761716134884405 -0.6974705366970222 2.8082311213145106
        children [
          DEF wheel Shape {
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 762.1437770537159
        anchor -0.045 0 0.025
      }
      device [
        RotationalMotor {
          name "motor_2"
        }
      ]
      endPoint Solid {
        translation -0.04505445655008966 -4.1614257457828526e-05 0.024976472323296577
        rotation -0.6287087941177594 -0.46058432543160743 -0.6265678984481283 4.0001808074700635
        children [
          USE wheel
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics USE wheel_physics
      }
    }
  ]
  boundingObject USE Body
  physics Physics {
  }
  controller "supervisor"
}
Apple {
  translation 0 0 0.15
  scale 2
}
Background {
  skyColor [
    1 1 1
  ]
}
